plugins/woocommerce/includes/rest-api/Controllers/Telemetry/class-wc-rest-telemetry-controller.php

```diff
@@ -109,10 +124,14 @@ public function get_usage_data( $request ) {
 			return;
 		}
 
+		// The installation date could be null from earlier mobile client versions.
+		$installation_date = $request->get_param( 'installation_date' );
+
 		return array(
-			'platform'  => sanitize_text_field( $platform ),
-			'version'   => sanitize_text_field( $version ),
-			'last_used' => gmdate( 'c' ),
+			'platform'          => sanitize_text_field( $platform ),
+			'version'           => sanitize_text_field( $version ),
+			'last_used'         => gmdate( 'c' ),
+			'installation_date' => get_gmt_from_date( $installation_date, 'c' ),
```

Comment:
I'm not sure what will happen if the `$installation_date` is `null`. It looks like it will [return a date regardless](https://developer.wordpress.org/reference/functions/get_gmt_from_date/). We probably want to return a `null` instead. 

I'm mentioning this because on the following code, we just copy over everything that is in `$new`. And I believe that includes `$new['installation_date']`. 🤔 So even if the client doesn't send `installation_date`, it will still be set? 

https://github.com/woocommerce/woocommerce/blob/71d47d6b4a16c380b42ddb9915f79c04243e10dc/plugins/woocommerce/includes/rest-api/Controllers/Telemetry/class-wc-rest-telemetry-controller.php#L97-L101

Maybe we also want to add a unit test that the endpoint behaves as expected if the iOS and Android clients are old (they only pass `version` and `platform`). Maybe confirm that `installation_date` is not set in that scenario. I think we don't have a test for this yet. But please let me know if I just missed it. 😅

Reply:
> So even if the client doesn't send installation_date, it will still be set?

Just tried this, and it's true. It just set the current date/time.

Reply:
Great catch on this case that I missed! I was able to reproduce it in a test case and API testing like Corey shared. Updated in https://github.com/woocommerce/woocommerce/pull/39605/commits/f22cff164d84606288492919c21eb4a41832bb8a for another look!